### PR TITLE
Added second part of unit tests for pkg/scheduler

### PR DESCRIPTION
**Pull request by anujagrawal699**
**Description:**
This PR significantly enhances the unit tests for the scheduler package and is the second part of unit tests for it. The improvements aim to increase code reliability, maintainability, and test coverage for various scheduler functions.

**Note:**
The CI tests might fail because the imports needed for tests are included in the first part of unit tests for scheduler module.

**Modifications:**
pkg/scheduler/scheduler_test.go

**Key Improvements:**
1. Added tests for scheduleClusterResourceBindingWithClusterAffinity and scheduleClusterResourceBindingWithClusterAffinities.
2. Added tests for the worker functionality and scheduleNext method.
3. Added tests for the placementChanged function.
4. Added new test cases for the TestCreateScheduler

**Test Coverage:**
pkg/scheduler/scheduler_test.go: 29.71% to 66.90%

**What type of PR is this?**
/kind feature

**Which issue(s) this PR fixes**:
Fixes a part of #5470 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by anujagrawal699** [comment]
@XiShanYongYe-Chang Please take a look.

File: pkg/scheduler/scheduler_test.go [hunk 1]
```diff
@@ -792,6 +796,420 @@ func TestPatchScheduleResultForResourceBinding(t *testing.T) {
...
+				},
+				Status: workv1alpha2.ResourceBindingStatus{
+					SchedulerObservedAffinityName: tt.observedAffinityName,
+				},
+			}
+			got := placementChanged(*crb.Spec.Placement, tt.appliedPlacementStr, crb.Status.SchedulerObservedAffinityName)
+			assert.Equal(t, tt.want, got, "placementChanged() for ClusterResourceBinding result mismatch")
+
+			// Test for ResourceBinding
+			rb := &workv1alpha2.ResourceBinding{
```
Comment (XiShanYongYe-Chang) [inline pkg/scheduler/scheduler_test.go:1195]: rb and crb should be no different here, is one enough to test?

Comment (anujagrawal699) [reply pkg/scheduler/scheduler_test.go:1195]: Yes, they have identical behavior.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**